backend/crates/eiffelvis_core/src/domain/user_queries.rs

```diff
@@ -0,0 +1,98 @@
+use crate::{
+    domain::types::BaseEvent,
+    graph::*,
+    tracked_query::{TrackedNodes, TrackedSubGraphs},
+};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Query {
+    /// Will return a given node if any of the filters match
+    filters: Vec<Filter>,
+    /// Method of collection, may add additional nodes
+    collection: Collection,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "type")]
+pub enum Filter {
+    /// Do not filter at all, just give everything
+    None,
+    /// Open ended range of time
+    Time {
+        begin: Option<u64>,
+        end: Option<u64>,
+    },
+    /// Event Type
+    Type { name: String },
+    /// Specific ids
+    Ids { ids: Vec<Uuid> },
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "type")]
+pub enum Collection {
+    /// Does not do anything
+    Forward,
+    /// Uses the result of the filtered nodes as roots and collects their graph
+    AsRoots,
+}
+
+pub struct TrackedQuery<I> {
+    filters: Vec<Filter>,
+    collector: Collector<I>,
+    inner: TrackedNodes<I>,
+}
```

Comment:
There should be a non tracked version of this so that we can support it as a single GET request (useful for debugging)